File: Controllers/HomeController.cs
using AuthorizationForm.Models;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Identity;
using Microsoft.AspNetCore.Mvc;

namespace AuthorizationForm.Controllers
{
    [AllowAnonymous]
    public class HomeController : Controller
    {
        private readonly UserManager<ApplicationUser> _userManager;
        private readonly SignInManager<ApplicationUser> _signInManager;
        private readonly ILogger<HomeController> _logger;

        public HomeController(
            UserManager<ApplicationUser> userManager, 
            SignInManager<ApplicationUser> signInManager,
            ILogger<HomeController> logger)
        {
            _userManager = userManager;
            _signInManager = signInManager;
            _logger = logger;
        }

        public async Task<IActionResult> Index()
        {
            // If user is not authenticated, redirect to login
            if (!User.Identity?.IsAuthenticated ?? true)
            {
                return RedirectToAction("Login", "Account");
            }

            var user = await _userManager.GetUserAsync(User);
            if (user == null) return RedirectToAction("Login", "Account");

            // Redirect Admin users to Admin panel
            if (await _userManager.IsInRoleAsync(user, "Admin"))
            {
                return RedirectToAction("Index", "Admin");
            }

            // Redirect Manager users to Manager dashboard
            if (await _userManager.IsInRoleAsync(user, "Manager"))
            {
                return RedirectToAction("Index", "Manager");
            }

            // Regular users go to requests page
            return RedirectToAction("Index", "Requests");
        }

        public IActionResult Error()
        {
            return View();
        }
    }
}

File: php-version/app/Http/Controllers/HomeController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\Auth;
use App\Models\User;

class HomeController extends Controller
{
    public function index()
    {
        // Check if setup is needed (for authenticated users accessing admin/management areas)
        if (Auth::check()) {
            $isSetupNeeded = $this->isSetupNeeded();
            if ($isSetupNeeded) {
                return redirect()->route('setup.index');
            }
        }

        // If user is not authenticated, redirect to create form (allow anonymous access)
        if (!Auth::check()) {
            return redirect()->route('requests.create');
        }

        $user = Auth::user();
        if (!$user) {
            return redirect()->route('login');
        }

        // Redirect Admin users to Admin panel
        if ($user->isAdmin()) {
            return redirect()->route('admin.index');
        }

        // Redirect Manager users to Manager dashboard
        if ($user->isManager()) {
            return redirect()->route('manager.index');
        }

        // Regular users go to requests page
        return redirect()->route('requests.index');
    }

    /**
     * Check if setup is needed
     */
    private function isSetupNeeded(): bool
    {
        // Check if any admin users exist
        $adminUsers = User::where('is_admin', true)->count();
        return $adminUsers === 0;
    }
}

